spec/01-unit/01-db/01-schema/11-snis_spec.lua

```diff
@@ -0,0 +1,90 @@
+local Schema = require "kong.db.schema"
+local snis = require "kong.db.schema.entities.snis"
+local certificates = require "kong.db.schema.entities.certificates"
+local utils = require "kong.tools.utils"
+
+Schema.new(certificates)
+local Snis = assert(Schema.new(snis))
+
+local function validate(b)
+  return Snis:validate(Snis:process_auto_fields(b, "insert"))
+end
+
+
+describe("snis", function()
+  local certificate = { id = utils.uuid() }
+
+  describe("name", function()
+    it("accepts a hostname", function()
+      local names = { "valid.name", "foo.valid.name", "bar.foo.valid.name" }
+
+      for _, name in ipairs(names) do
+        local ok, err = validate({ name = name, certificate = certificate })
+        assert.is_nil(err)
+        assert.is_true(ok)
+      end
+    end)
+
+    it("accepts wildcards", function()
+      local names = { "*.wildcard.com", "wildcard.*", "test.wildcard.*",
+                      "foo.test.wildcard.*", "*.test.wildcard.com" }
+
+      for _, name in ipairs(names) do
+        local ok, err = validate({ name = name, certificate = certificate })
+        assert.is_nil(err)
+        assert.is_true(ok)
+      end
+    end)
+
+    it("rejects wrong wildcard placements", function()
+      local names = { "foo.*.com", "foo.*.wildcard.com" }
+
+      for _, name in ipairs(names) do
+        local ok, err = validate({ name = name, certificate = certificate })
+        assert.is_nil(ok)
+        assert.same({ name = "wildcard must be leftmost or rightmost character" }, err)
```

Comment:
Changes needed for the added cases:
```suggestion
        assert.is_true(err.name == "wildcard must be leftmost or rightmost character"
                       or err.name == "only one wildcard must be specified")
```